p2p/server/server_test.go

```diff
@@ -32,12 +34,23 @@ func TestServer(t *testing.T) {
 	}
 	opts := []Opt{
 		WithTimeout(100 * time.Millisecond),
-		WithContext(ctx),
+		WithLog(logtest.New(t)),
 	}
 	client := New(mesh.Hosts()[0], proto, handler, append(opts, WithRequestSizeLimit(2*limit))...)
-	_ = New(mesh.Hosts()[1], proto, handler, append(opts, WithRequestSizeLimit(limit))...)
-	_ = New(mesh.Hosts()[2], proto, errhandler, append(opts, WithRequestSizeLimit(limit))...)
-
+	srv1 := New(mesh.Hosts()[1], proto, handler, append(opts, WithRequestSizeLimit(limit))...)
+	srv2 := New(mesh.Hosts()[2], proto, errhandler, append(opts, WithRequestSizeLimit(limit))...)
+	ctx, cancel := context.WithCancel(context.Background())
+	var eg errgroup.Group
+	eg.Go(func() error {
+		return srv1.Run(ctx)
+	})
+	eg.Go(func() error {
+		return srv2.Run(ctx)
+	})
+	t.Cleanup(func() {
+		cancel()
+		eg.Wait()
+	})
```

Comment:
I don't see how a blocking `Run` method is more convenient to use than something like this:

```suggestion
	require.NoError(t, srv1.Start())
	t.Cleanup(t, func { assert.NoError(t, srv1.Stop()) })
	require.NoError(t, srv2.Start())
	t.Cleanup(t, func { assert.NoError(t, srv2.Stop()) })
```

Reply:
it is convenient to integrate in the App. additionally if any of the Run fails with error, all other runs should fail as well, doing this with Start/Stop is not clean